src/main/java/winstone/Launcher.java

```diff
@@ -65,6 +68,8 @@ public class Launcher implements Runnable {
 
     private int CONTROL_TIMEOUT = 2000; // wait 2s for control connection
 
+    private static int SHUTDOWN_TIMEOUT = 5000; // wait 5s for shutdown
```

Comment:
With a lot of plugins running `@Terminator`s, 5s might be tight. I would suggest 10–20s as a more typical shutdown grace period for Jenkins.

Reply:
I'll change it to 20 seconds